dao/mongodb.rya/src/main/java/mvm/rya/mongodb/MongoConnectorFactory.java

```diff
@@ -41,26 +47,77 @@
      * @param conf The {@link Configuration} defining how to construct the MongoClient.
      * @return A {@link MongoClient}.  This client is lazy loaded and the same one
      * is used throughout the lifecycle of the application.
-     * @throws NumberFormatException - Thrown if the configured port is not a valid number
-     * @throws UnknownHostException - The configured host cannot be found.
+     * @throws IOException - if MongodForTestsFactory constructor has an io exception.
+     * @throws ConfigurationRuntimeException - Thrown if the configured server, port, user, or others are missing.
+     * @throws UnknownHostException - if MongoDB host name is not found.
      */
-    public static synchronized MongoClient getMongoClient(final Configuration conf) throws NumberFormatException, UnknownHostException {
-        if(mongoClient == null) {
-            final String host = conf.get(MongoDBRdfConfiguration.MONGO_INSTANCE);
-            final int port = Integer.parseInt(conf.get(MongoDBRdfConfiguration.MONGO_INSTANCE_PORT));
-            final ServerAddress server = new ServerAddress(host, port);
-
-            //check for authentication credentials
-            if (conf.get(MongoDBRdfConfiguration.MONGO_USER) != null) {
-                final String username = conf.get(MongoDBRdfConfiguration.MONGO_USER);
-                final String dbName = conf.get(MongoDBRdfConfiguration.MONGO_DB_NAME);
-                final char[] pswd = conf.get(MongoDBRdfConfiguration.MONGO_USER_PASSWORD).toCharArray();
-                final MongoCredential cred = MongoCredential.createCredential(username, dbName, pswd);
-                mongoClient = new MongoClient(server, Arrays.asList(cred));
+    public static synchronized MongoClient getMongoClient(final Configuration conf)
+            throws ConfigurationRuntimeException, MongoException {
+        if (mongoClient == null) {
+            // The singleton client has not yet created, is it a test/mock instance, or a service?
+            if (conf.getBoolean(MongoDBRdfConfiguration.USE_TEST_MONGO, false)) {
+                try {
+                    MongodForTestsFactory testsFactory = MongodForTestsFactory.with(Version.Main.PRODUCTION);
+                    mongoClient = testsFactory.newMongo();
+                } catch (IOException e) {
+                    // Rethrow as an unchecked error.  Since we are in a test mode here, just fail fast.
+                    throw new MongoException("While creating a factory for a test/mock MongoDB instance.",e);
+                }
             } else {
-                mongoClient = new MongoClient(server);
+                // Connect to a running Mongo server
+                final String host = requireNonNull(conf.get(MongoDBRdfConfiguration.MONGO_INSTANCE), "MongoDB host name is required");
```

Comment:
I started with that.  (Objects.requireNonNull() is similar)   But I wanted to throw a different exception, not NullPointerException since the caller may want to specifically catch it.  So then I wrapped the whole thing in a try with a catch that re-throws.  But then it catches ANY null pointer, not just the ones I checked and it became more complicated, so I did it this way.  What do you think, is it worth it to throw a different unchecked error?